pilot/pkg/bootstrap/server.go

```diff
@@ -431,11 +431,10 @@ func (s *Server) WaitUntilCompletion() {
 
 // initSDSServer starts the SDS server
 func (s *Server) initSDSServer(args *PilotArgs) {
-	if features.EnableSDSServer && s.kubeClient != nil {
+	if s.kubeClient != nil {
 		if !features.EnableXDSIdentityCheck {
 			// Make sure we have security
-			log.Warnf("skipping Kubernetes credential reader, which was enabled by ISTIOD_ENABLE_SDS_SERVER. " +
-				"PILOT_ENABLE_XDS_IDENTITY_CHECK must be set to true for this feature.")
+			log.Warnf("skipping Kubernetes credential reader; PILOT_ENABLE_XDS_IDENTITY_CHECK must be set to true for this feature.")
 		} else {
```

Comment:
Do we still need this if we remove the local gateway SDS server ? And one step ahead - can we remove the dep on k8s library ?

Reply:
I think so? we still need a way to ensure we don't give certs to anyone who asks. We should revive the "sds auth" work.

For the k8s dep - I have a branch with it removed but I am slowly sending it over many PRs since it requires many many changes. End result is 1/3 size of agent binary